drivers/dac/dac_stm32.c

```diff
@@ -0,0 +1,180 @@
+			table_channels[channel - STM32_FIRST_CHANNEL], value);
+	} else {
+		LOG_ERR("Invalid resolution");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dac_stm32_channel_setup(struct device *dev,
+				   const struct dac_channel_cfg *channel_cfg)
+{
+	struct dac_stm32_data *data = dev->driver_data;
+	const struct dac_stm32_cfg *cfg = dev->config->config_info;
+
+	if (channel_cfg->channel_id - STM32_FIRST_CHANNEL >=
+			data->channel_count ||
+			channel_cfg->channel_id < STM32_FIRST_CHANNEL) {
+		LOG_ERR("Channel %d is not valid", channel_cfg->channel_id);
+		return -EINVAL;
+	}
+
+	/*
+	 * STM32 MCUs either use VDDA as the reference (e.g. F0 series) or a
+	 * dedicated VREF+ pin. The VREF+ pin can be connected to an internal
+	 * reference (shared with ADC), driven by an external reference or
+	 * wired to VDD.
+	 *
+	 * In any case, the reference of the DAC cannot be selected (at least
+	 * not independent of the ADC). So we only accept ADC_REF_DEFAULT.
+	 */
+	if (channel_cfg->reference != DAC_REF_DEFAULT) {
+		LOG_ERR("Invalid channel reference");
+		return -EINVAL;
+	}
+
+	LL_DAC_SetOutputBuffer(cfg->base,
+		table_channels[channel_cfg->channel_id - STM32_FIRST_CHANNEL],
+		channel_cfg->buffered ? LL_DAC_OUTPUT_BUFFER_ENABLE :
+		LL_DAC_OUTPUT_BUFFER_DISABLE);
+
+	LL_DAC_Enable(cfg->base,
+		table_channels[channel_cfg->channel_id - STM32_FIRST_CHANNEL]);
+
+	LOG_DBG("Channel setup succeeded!");
+
+	return 0;
+}
+
+static int dac_stm32_init(struct device *dev)
+{
+	const struct dac_stm32_cfg *cfg = dev->config->config_info;
+
+	/* enable clock for subsystem */
+	struct device *clk =
+		device_get_binding(STM32_CLOCK_CONTROL_NAME);
+
+	if (clock_control_on(clk,
+			     (clock_control_subsys_t *) &cfg->pclken) != 0) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static const struct dac_driver_api api_stm32_driver_api = {
+	.channel_setup = dac_stm32_channel_setup,
+	.write_value = dac_stm32_write_value,
+	/*
+	 * No independent internal reference available for DAC peripheral,
+	 * DAC uses voltage reference from VDDA or VREF+ pins that may be set by
+	 * external hardware.
+	 */
+	.ref_internal = 0
+};
+
+
+#define STM32_DAC_INIT_INST(index)					\
+									\
+static const struct dac_stm32_cfg dac_stm32_cfg_##index = {		\
+	.base = (DAC_TypeDef *)						\
+		DT_INST_##index##_ST_STM32_DAC_BASE_ADDRESS,		\
+	.pclken = {							\
+		.enr = DT_INST_##index##_ST_STM32_DAC_CLOCK_BITS,	\
+		.bus = DT_INST_##index##_ST_STM32_DAC_CLOCK_BUS,	\
+	},								\
+};									\
+static struct dac_stm32_data dac_stm32_data_##index = {			\
+	.channel_count = STM32_CHANNEL_COUNT				\
+};									\
+									\
+DEVICE_AND_API_INIT(dac_##index, DT_INST_##index##_ST_STM32_DAC_LABEL,	\
+		    &dac_stm32_init, &dac_stm32_data_##index,		\
+		    &dac_stm32_cfg_##index, POST_KERNEL,		\
+		    CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,		\
+		    &api_stm32_driver_api)
+
+/* DT instance numbering starts from 0, STM DAC from 1 */
+
+#ifdef CONFIG_DAC_1
```

Comment:
Change these from `#ifdef CONFIG_DAC_1` to `#ifdef DT_INST_0_ST_STM32_DAC`

Reply:
Done.